lib/trilogy_adapter/errors.rb

```diff
@@ -1,5 +1,12 @@
 # frozen_string_literal: true
 
+module ActiveRecord
+  if version < ::Gem::Version.new('7.1.a')
+    class ConnectionFailed < QueryAborted
+    end
+  end
+end
```

Comment:
won't it be better to conditionally define the base class instead of monkey patching AR ?

Reply:
Probably so. Have placed this change in a separate commit in case there are any different ideas about how to implement these error updates.